### ISSUE TITLE
Sometimes participants can't hear other participants

### ISSUE THREAD

**Issue by jonnius** [jitsi/jitsi-meet#5230]
## Description
Sometimes some participants can't hear some other participants. Some findings:
* occurs more often in meetings with many participants (does not seem to occur with 4 participants, occurs a lot with >=7 participants)
* does not happen between users with camera enabled
* is not necessarily bidirectional (A might hear B, while B does not hear A)
* workarounds
  * in most cases one can work around the issue by toggling the camera
  * in some cases one can work around the issue by reloading the page
  * in some cases none of the workarounds above helps
* issue seems to be around for years already, as it might be a duplicate to the closed issues #1278 and #641

## Current behavior
There is an audio-only meeting with n>7 participants. Some participants cannot hear all other participants, e.g. participant 4 cannot hear participant 3 and 7.

## Expected Behavior
Every participant can hear every other participant.

## Possible Solution
---

## Steps to reproduce
Start a Jitis Meeting with as many users as possible, only enable Audio and try to speak to each other.

# Environment details
Version: jitsi-meet v1.0.4101-1
Browsers tested: Chrome, Firefox on Win10 and Ubuntu 16.04

**Comment by rettichschnidi** [comment]
Pretty sure this is related to #4758 

Also, other users (including me) have the same problem:
* https://community.jitsi.org/t/jitsi-users-use-meet-jit-si-with-many-users/13187
* https://community.jitsi.org/t/not-everyone-hear-all-participants/23986
* https://community.jitsi.org/t/meet-jit-si-app-meeting-participants-end-up-in-two-separate-meetings/23253

**Comment by damencho** [comment]
Is Firefox involved?

**Comment by rettichschnidi** [comment]
Yes

**Comment by damencho** [comment]
We are working on improving Firefox experience, but that is for now. Firefox sometimes does not send or receive any streams ... and leads to problems. That's why we put a notification when you are on Firefox and using meet.jit.si, to warn people.

**Comment by jonnius** [comment]
We experienced the issues also on Chrome. 

We just updated to jitsi-meet v1.0.4296-1 and disabled Peer-to-Peer connections. On a test with 10 participants the issue did not occur on Chrome any more. On a test with everyone on Firefox 1 out of 10 could not hear some others.

**Comment by damencho** [comment]
We will update very soon stable.

**Comment by Jasper-Ben** [comment]
We are experiencing the same issue with 9 people in the same room, even between users using chromium-based browsers. We also tried to connect with all participants using said browsers (which I'm 90% sure we archived) and issues still persisted. So maybe not just a firefox issue?

**Comment by jonnius** [comment]
We did as well before updating. Which version are you using?

**Comment by Jasper-Ben** [comment]
the latest jitsi-meet package currently available from jitsis debian repository: jitsi-meet/stable,now 1.0.4101-1

It used to work when microphone and cam where enabled by default when joining a room during our testing. But since we disabled that in the jitsi settings it is completely unreliable. So for me it seems the issue is related to the (lack of) video or audio streaming.
Out of privacy concerns I am unfortunately unable to reactivate said settings to see if it makes any difference, as 20+ users in our company now expect it to be disabled by default.

Also I was able to reproduce the issue in a two-participants room (although one was using firefox), so this issue does not seem to be bound to the number of users in a room as suggested in the initial report. There might have been users in other rooms at that time though, so it might be a question of how many users are connected, regardless of room.

**Comment by jonnius** [comment]
You are using the very same version we experienced this issue a lot with. After updating to the 4296 build we did not see this issue on Chrome any more. We also have mic and cam disable by default.

**Comment by Jasper-Ben** [comment]
thanks for the info! will try the nightly branch then and install the same version. I will report back, if we continue to experience issues.

**Comment by jonnius** [comment]
We just had a meeting with 11 participants, all on Chrome, where one participant could not be heard by the others. Reloading the page for the one fixed it.

**Comment by saghul** [comment]
Thanks for chiming in @jonnius !

**Comment by Jasper-Ben** [comment]
We just had a test meeting with 20 participants (all on chromium) and it worked flawless. Also build 4296. Seems to be fixed :+1:

**Comment by Jasper-Ben** [comment]
P.S.: we actually had one minor issue with a participant who used Vivaldi. After he switched to chrome it worked as expected. We didn't investigate further but my guess would be that Vivaldi might use an old chromium base. Just something to be aware of.

**Comment by Isolator70** [comment]
Hello, using Jitsi sporadically, but due to Corona crisis now more often.
Today I had a 2 person call between Chromium@LinuxMint_18.3_PC and Jitsi App@Android. Everything fully up-to-date. After some minutes it was not possible anymore for the Person at Android to hear the Person from the PC.
At the PC one could observe, that the microphone indicator did not react to volume put into the microphone.
Going to the settings of Jitsi in Chromium browser one could see "Permission not granted" for the audio stuff (although it was done and checked at the beginning of the meeting).
Checking the audio permission in Chromium itself, camera and audio stuff was enabled.
After closing and restarting the meeting everything was fine again, but broke down some minutes later again.
Do you think it is the same issue or shall I create another ticket?
Thank you

**Comment by Isolator70** [comment]
Okay, I have to add following information: Today I was in a 35 minutes video call with up to 17 participants, most of the without activated camera, 5 with camera activated. I'm sure most of them were on Windows machines, 1 maybe on Mac, 1 with iPhone, 1 with Android,
From my point of view, everything ran perfect - I couldn't see the issue of yesterday again! So I will observe the behaviour if it is random.

**Comment by keinstein** [comment]
Hi, on March 27th, we had a meeting with 3 people on meet.jit.si (I don't know which version they run). All using web browsers and we had the problem that the 3rd person connected but did not receive any audio or video. After he managed to setup his audio and video settings, person 1 could hear both 2 and 3, 2 could hear only 1, 3 only 2. So every one was streaming something to someone. But somehow the streams stopped working.

**Comment by real-or-random** [comment]
We observed this today in 3-person call on 4384. A Jigasi/SIP peer couldn't hear an electron peer. Everything else worked. "Restarting" the stream by changing the microphone device on electron and changing back fixed this.

**Comment by thomaswucher** [comment]
We also observed this problem today on a self-hosted instance using the latest official docker images and the master branch of docker-jitsi-meet (0177765f) in a conference with 5 participants.

All participants were using Firefox. The ones which had video enabled didn't experience the problem, those with audio only could sometimes not hear all other participants. For those experiencing problems, reconnecting sometimes helped.

**Comment by Jasper-Ben** [comment]
> All participants were using Firefox.

Hi Thomas, as things stand right now I'd suggest you enforce a policy of chromium based browsers only, as issues with Firefox are still being resolved. And although the related issue https://github.com/jitsi/jitsi-meet/issues/4758 is marked as resolved, this to my understanding only fixes any issues on the jitsi side of things. There are still open upstream issues in Firefox (linked at the bottom of the thread).

We enforce our chromium only policy by setting the following in the `interface_config.js`:
```
OPTIMAL_BROWSERS: [ 'chrome', 'chromium', 'nwjs', 'electron' ],
UNSUPPORTED_BROWSERS: ['firefox','safari'],
```
This causes users trying to open jitsi with a Firefox or Safari Browser to be prompted to install the latest Google Chrome. For anyone who doesn't want to use Chrome out of privacy concerns we recommend using Chromium or Brave, as they are both open-source and (for the most part) do not include any Google tracking stuff.

**Comment by jonnius** [comment]
Things should improve with Firefox 76.

**Comment by btsimonh** [comment]
I'm currently investigating a problem with jitsi which sounds very much like this and previous descriptions.

We had 3 people on the call, All Chrome 81.0.4044.129 on windows 10 64 bit using embedded jitsi from 'https://meet.jit.si/external_api.js'. - So this may highlight that it's not a firefox only issue, and hopefully may provide some more pointers....
Certainly the 'unheard' person was sending audio, because they were heard by another, yet not heard by the third person.  The 'unheard' person could hear the person who could not hear them....  The person who could not hear them could hear the other person.

Prior to integrating, we tested 7-10 calls using meet.jit.si for bulk family calls (always with video?), and apart from once on MAC which didn't screenshare whatever we did (it had previously), we had no (apparent) problems.
 
I find this in the browser log as a yellow:
`Logger.js:154 2020-05-04T15:56:33.440Z [modules/statistics/AudioOutputProblemDetector.js] A potential problem is detected with the audio output for participant 02b26f22, local audio levels: [null,null], remote audio levels: undefined`

If I separately join the room using meet.jit.si/roomname, then I too could hear the person, but not through the embedded version (could i just have been lucky?).  But this was not an extensive test (3 times, it worked).

I start all the instances like this:
(i.e. 
video muted; which could be related/
these ones because they mess with my audio mojo (i.e. I'm also recording quality audio from the mic, and these seem to mess with Chrome's global mic gain).
enableNoisyMicDetection:false, - I don't need or want the 'pip-pip-pip' indication by audio...
enableNoAudioDetection:false,
enableTalkWhileMuted:false,
disableAudioLevels: true,
)
```
            let config = {
              enableNoisyMicDetection:false,
              enableNoAudioDetection:false,
              enableTalkWhileMuted:false,
              disableAudioLevels: true,

              // whole room by creator?
              //startAudioOnly:true,
              startWithAudioMuted: true,
              // only this person
              startWithVideoMuted: true,
              enableCalendarIntegration: false,
              disableThirdPartyRequests: true,
              disableRecordAudioNotification: true,
              //testing:{
                //noAutoPlayVideo:true,
              //},
              //chromeExtensionBanner: {},
              disableInviteFunctions: true,
              disableRemoteMute: true,
            };

            let interfaceConfig = {
              DISPLAY_WELCOME_PAGE_CONTENT: false,
              MOBILE_APP_PROMO: false,


            };

            this.jitsi_api = new JitsiMeetExternalAPI('meet.jit.si', {
              parentNode: this.$refs['putithere'],
              roomName: this.settings.roomName,
              configOverwrite: config,
              interfaceConfigOverwrite: interfaceConfig,
            });

```

it's important I work around this in the very short term; any ideas?

Simon

**Comment by Jasper-Ben** [comment]
> video muted; which could be related/

In my experience the issue described here only occurs when disabling audio on entering the room.
So changing `startWithAudioMuted: true,` to `startWithAudioMuted: false,` might be a short term workaround?

Other than that, I cannot really help as I am unfamiliar with the api.

**Comment by btsimonh** [comment]
something else; watch out for
The AudioContext was not allowed to start. It must be resumed (or created) after a user gesture on the page. https://goo.gl/7K7WLu
I think this prevented one of my jitsi clients from starting audio?

**Comment by tomkel** [comment]
I still have this bug in rel 4548

**Comment by luixxiul** [comment]
> 
> 
> I still have this bug in rel 4548. In a call with ~15 people

As this issue has been closed and the thing has been changed since then, there is actually nothing actionable here. You are expected to open a new issue, with steps to reproduce the error, server / client information, etc.

**Comment by spaquet** [comment]
> Is Firefox involved?

I have the same issue on Chrome

**Comment by dandv** [comment]
> DISPLAY_WELCOME_PAGE_CONTENT: false,

@btsimonh, I see you've configured `DISPLAY_WELCOME_PAGE_CONTENT`. Have you figured out [what it does](https://github.com/jitsi/jitsi-meet/issues/7180)?

**Comment by tomkel** [comment]
@luixxiul That's easier said than done, it's an intermittent issue and it usually takes 10-15 clients before the issue starts happening. What kind of logs are needed and how do I get them?

**Comment by btsimonh** [comment]
@dandv - no.  I just scanned the code for all possible configurations, and threw in random ones :).
@tomkel - I regularly have sessions with 3-4 clients where one client can't hear another, but others can hear them.  It seems like a very complex issue, and probably multiple subtle issues.  I'd love to have it happen when I have easy access to the clients in question, so I could try to diagnose, but it's rather difficult.
My application is a simple audio only intercom, and it fails regularly enough for me to be moving away from JitSi to a much simpler audio intercom over websockets with dedicated self-hosted servers.  This also saves me a load of CPU, but costs me a load of dev time :(.  I may retain JitSi, but only as a link to be used when screenshare is useful (it's a collaborative app, where some need help setting up).

**Comment by Nicramus** [comment]
> something else; watch out for
> The AudioContext was not allowed to start. It must be resumed (or created) after a user gesture on the page. https://goo.gl/7K7WLu
> I think this prevented one of my jitsi clients from starting audio?

good point - in my case that was the problem

**Comment by loelkes** [comment]
I still have this issue with the most recent release. Can someone point out what qualifies as "user gesture"? Must the page be reloaded after a user gesture?

**Comment by damencho** [comment]
Using pre-join page and there shouldn't be a problem. It needs a gesture or webrtc session before playing it is how Google changed it ... not sure about FF though.

**Comment by arianox** [comment]
We have had the problem since the last release too. But very strange as it only occurs sometimes.

**Comment by damencho** [comment]
What is the browser and version you experience the problem with?

**Comment by arianox** [comment]
> What is the browser and version you experience the problem with?

- Chrome (latest version)
- Microsoft Edge (latest version)

**Comment by damencho** [comment]
It would be helpful if you can get the js console logs from a participant that is not hearing someone, but also note which participant is not heard (by checking its participant id from the extended local sstats)
![image](https://user-images.githubusercontent.com/3263098/117023521-cc858c00-acbe-11eb-98cf-c390c3907799.png)

And note is the participant not hearing from the beginning of the session or stopped hearing at some point...

**Comment by jallamsetty1** [comment]
@arianox, Can you please share the full log if possible ? The logs attached in your last comment do not show anything wrong with the source addition. You can ignore the warning from AudioOutputProblemDetector, they are misleading and do not indicate a real problem with remote audio.

**Comment by jallamsetty1** [comment]
> @jallamsetty1 OK. I will try to get the data at the next conference.

Thank you, can you please make a note of the endpointId for the user that wasn't heard ? You can find this information on the connection stats on the thumbnail for the remote participant.

**Comment by ssb22** [comment]
We had this problem this morning on a 3-way conversation where A could hear only C but B and C could hear each of the other two.  The issue persisted over two separate sessions of the same meeting room.

Participant A was using Safari on an iPhone SE from 2016 running iOS 14 with all updates accepted.  Jitsi app was not installed because the phone was getting full, so Safari was used.  Cameras were off as participants were blind (it's risky to turn on a camera when you can't see what it's pointing at).

**Comment by Ark74** [comment]
I've seen that with firefox if you are the first on the room you don't have issues.

If you are 2nd or later **and** last, then you'll have sound issues.
If you are 2nd or later, you'll have issues 'til another user join the room, that kind of triggers sound discovery.

**Comment by ssb22** [comment]
Well one of the things we tried was to have participant B leave the meeting and re-join, but this did not trigger participant A's Safari to work any better.

**Comment by jallamsetty1** [comment]
The audio issues with Safari and Firefox have already been fixed and will be deployed to meet.jit.si early this week.

**Comment by jamesbrooks** [comment]
> The audio issues with Safari and Firefox have already been fixed and will be deployed to meet.jit.si early this week.

Awesome! Looking forward to seeing this fix merged in :)

**Comment by jallamsetty1** [comment]
> > The audio issues with Safari and Firefox have already been fixed and will be deployed to meet.jit.si early this week.
> 
> Awesome! Looking forward to seeing this fix merged in :)

The fix has been deployed on meet.jit.si already. Please give it a try and let me know if it fixes your issue, thanks!

**Comment by domrim** [comment]
@jallamsetty1 we have the same issues as everyone else here. Is this fix allready present in jitsi/stable? or better: which commit fixed the issue?

**Comment by loelkes** [comment]
Hi Dominik ;)

> @jallamsetty1 we have the same issues as everyone else here. Is this fix allready present in jitsi/stable? or better: which commit fixed the issue?

Check your JavaScript console logs on the clients facing the audio issues. If there are AudioContext errors than try enabling the pre-join page. If the browser enforces the requirement for an user gesture before accessing the AudioContext it seems there's no way around the pre-join page. It seems a bit random how strictly browser implement this.

Enabling the Pre-Join page solved an incredible amount of issues related to audio on all my setups.

**Comment by c0m1t** [comment]
We are still experiencing the same issue, mainly on Chrome.
This is the error:
```
Error: replace track failed
```
Seems like `_findTransceiver` in `lib-jitsi-meet` returns `null`.

**Comment by midn1** [comment]
Still happens on meet.ji.si, and it is very difficult to keep people from Zoom when this sort of thing is common..

**Comment by c0m1t** [comment]
> Using pre-join page and there shouldn't be a problem. It needs a gesture or webrtc session before playing it is how Google changed it ... not sure about FF though.

@damencho We have a prejoin page but sometimes on chrome browser on mobile we get this error:

```
​ The AudioContext was not allowed to start. It must be resumed (or created) after a user gesture on the page. https://goo.gl/7K7WLu
(anonymous) @ lib-jitsi-meet.min.js:1
n @ lib-jitsi-meet.min.js:1
(anonymous) @ lib-jitsi-meet.min.js:1
(anonymous) @ lib-jitsi-meet.min.js:1
n @ lib-jitsi-meet.min.js:1
(anonymous) @ lib-jitsi-meet.min.js:10
(anonymous) @ lib-jitsi-meet.min.js:10
n @ lib-jitsi-meet.min.js:1
(anonymous) @ lib-jitsi-meet.min.js:10
(anonymous) @ lib-jitsi-meet.min.js:10
n @ lib-jitsi-meet.min.js:1
(anonymous) @ lib-jitsi-meet.min.js:10
n @ lib-jitsi-meet.min.js:1
(anonymous) @ lib-jitsi-meet.min.js:17
(anonymous) @ lib-jitsi-meet.min.js:17
n @ lib-jitsi-meet.min.js:1
(anonymous) @ lib-jitsi-meet.min.js:17
n @ lib-jitsi-meet.min.js:1
(anonymous) @ lib-jitsi-meet.min.js:1
(anonymous) @ lib-jitsi-meet.min.js:1
(anonymous) @ lib-jitsi-meet.min.js:1
(anonymous) @ lib-jitsi-meet.min.js:1
lang/countries-fa.json?r=1638779169516:1 Failed to load resource: the server responded with a status of 404 ()
​ [Deprecation] The ScriptProcessorNode is deprecated. Use AudioWorkletNode instead. (https://bit.ly/audio-worklet)
_initializeAudioContext @ lib-jitsi-meet.min.js:1
c @ lib-jitsi-meet.min.js:1
(anonymous) @ lib-jitsi-meet.min.js:1
```

When this happens other participants cannot hear the participant audio.

**Comment by de1phi48** [comment]
Good day, we have a same problem. in a conference of 15 people, one or two people may not be able to hear the others, and the microphone stops working. We use own instanse Jitsi+Rocket.Chat

**Comment by havaeng** [comment]
Hello, just wanted to inform that I'm experiencing this right now. I can't hear 2 of the 7 participants (not counting myself). Everyone can hear me. Safari version 17.0 (19616.1.27.211.1)

**Comment by saghul** [comment]
@jallamsetty1 ^

**Comment by jallamsetty1** [comment]
Hi @havaeng, can you please share more details? Is this happening on meet.jit.si or on your deployment? Were you able to hear those 2 users in the beginning and then stopped hearing them all of a sudden or did the issue exist from the moment those users unmuted themselves and started speaking? Will you be able to share the full browser console logs for the call when this issue reproduces?

**Comment by isality** [comment]
Same issue.

Firefox 128.0
Jitsi stable-9457

no sound and video in Firefox. Chrome works fine.

I got error messages in console:

```
2024-07-16T08:38:31.726Z [modules/statistics/AudioOutputProblemDetector.js] <526/_onLocalAudioLevelsReport/<>:  A potential problem is detected with the audio output for participant 63c60420, local audio levels: [null,null], remote audio levels: undefined [Logger.js:155:26](webpack://JitsiMeetJS/node_modules/@jitsi/logger/lib/Logger.js)
2024-07-16T08:38:31.727Z [modules/statistics/AudioOutputProblemDetector.js] <526/_onLocalAudioLevelsReport/<>:  A potential problem is detected with the audio output for participant becfffa0, local audio levels: [null,null], remote audio levels: undefined [Logger.js:155:26](webpack://JitsiMeetJS/node_modules/@jitsi/logger/lib/Logger.js)
2024-07-16T08:38:31.727Z [modules/statistics/AudioOutputProblemDetector.js] <526/_onLocalAudioLevelsReport/<>:  A potential problem is detected with the audio output for participant 3f6da3ec, local audio levels: [null,null], remote audio levels: undefined [Logger.js:155:26](webpack://JitsiMeetJS/node_modules/@jitsi/logger/lib/Logger.js)
2024-07-16T08:38:31.727Z [modules/statistics/AudioOutputProblemDetector.js] <526/_onLocalAudioLevelsReport/<>:  A potential problem is detected with the audio output for participant e3aee918, local audio levels: [null,null], remote audio levels: undefined
```


Could someone check that?

**Comment by saghul** [comment]
CAn you reproduce this consistently? Can you please share the full JS console logs?

**Comment by isality** [comment]
@saghul  hi, full console log in attachment.

stepps to reproduce:

1. go to any rooms
2. no one hears what I say
3. I can't hear anyone
4.  web camera does not work also

if im changed browser on google chrome, evrything works fine 

[Uploading console-export-2024-7-16_16-39-23.txt…]()

**Comment by saghul** [comment]
The link seems to be broken.

**Comment by isality** [comment]
@saghul link fixed

**Comment by jallamsetty1** [comment]
Can you please try with the latest stable? There were some breaking changes with Firefox and DTLS 1.3 that have been addressed recently in the JVB.

**Comment by isality** [comment]
yes, I can, but it will take a lot of time. The update process is very slow in my company.

**Comment by jallamsetty1** [comment]
Can you reproduce the issue with older versions of Firefox? If not, then the DTLS change in latest FF is the likely culprit and updating the JVB is the only option then.

**Comment by jallamsetty1** [comment]
> Firefox 125 same. No sound / video. Tested on mac pro (m2) and ubuntu 22.04.

Are you able to reproduce this on meet.jit.si?

**Comment by isality** [comment]
meet.jit.si - works fine!

**Comment by isality** [comment]
@jallamsetty1 hi! whats version of jvb server and jitsi using meet.jit.si?

**Comment by saghul** [comment]
That deployment is always a bit ahead of our stable releases, but in your case using the latest stable release should do it.

**Comment by isality** [comment]
@saghul  thank you. I think the update will solve my problem.

**Comment by isality** [comment]
after update on latest stable release - firefox works fine.

**Comment by kiwi2kiwi** [comment]
I just had the same issue on the latest firefox build. 13 people in a meeting and i wasn't able to hear 2 people, but they could hear me and others heard them as well

**Comment by Ark74** [comment]
I've noticed this only when users are on different OS, like users running MacOS + Google Chrome with other ones on Ubuntu + Firefox or Windows + Edge / Chrome.

Only when all users passed to Chrome based ones, sound actually got in sync and the session could go without issues.

I could be wrong, but from my experience Windows + Ubuntu (/ derivatives) have no issues even when there is a Chrome + Firefox mix, only when MacOS users join is that all users need to be on Chrome based browsers, so maybe an issue handling vp8/vp9 among OS's.

Regards.

**Comment by optimuswhite** [comment]
Is there a solution for this problem? Just yesterday, in a meeting I attended, I could not be heard even though there was no problem with my devices and permissions. My console log was as follows. 

By the way, I was able to solve the problem by restarting my computer. 

www-slwpp3ez:1 The AudioContext encountered an error from the audio device or the WebAudio renderer.
Logger.js:154 2025-03-10T06:10:24.363Z [modules/RTC/JitsiLocalTrack.js] LocalTrack[2,audio] 'bytes sent' <= 0: 0
Logger.js:154 2025-03-10T06:10:24.363Z [modules/RTC/JitsiLocalTrack.js] LocalTrack[2,audio] 'bytes sent' <= 0: 0
